### ISSUE TITLE
Incorrect Column Name in tessellate_subduction_zones Function

### ISSUE THREAD

**Issue by singhsatyampratap** [GPlates/gplately#270]
**Description:**

In the `PlateReconstruction.tessellate_subduction_zones(return_geodataframe=True)` function, the returned DataFrame includes a column named `overriding_plate_id`. However, this column should be named `trench_plate_id` to accurately reflect the data it represents. The current `overriding_plate_id` values differ significantly from the trench plate IDs, as shown in the attached figure.

**Steps to Reproduce:**

1. Use the `PlateReconstruction.tessellate_subduction_zones(return_geodataframe=True)` function.
2. Inspect the returned DataFrame.

Please refer to the attached figure for a visual comparison of the `overriding_plate_id` and `trench_plate_id`.

**Attachment:**
![Trench_Plate_ID](https://github.com/user-attachments/assets/991cd29a-fe1a-438f-a1fb-0427f7dcfeb2)

**Comment by jcannon-gplates** [comment]
Yes, that's true, it probably should be changed to "trench plate ID" [here](https://github.com/GPlates/gplately/blob/e9bcbcd5f16d075a1746fe47d66b8de5eaafd525/gplately/reconstruction.py#L270). I don't think that'll cause a problem - if I don't hear from any @GPlates/gplates-dev then I'll change that next week.

Originally the subduction convergence script (previously in PlateTectonicTools) used to find the plate ID of the overriding plate, but was subsequently changed to the plate ID of the trench line for [this reason](https://github.com/GPlates/gplately/blob/e9bcbcd5f16d075a1746fe47d66b8de5eaafd525/gplately/ptt/subduction_convergence.py#L402-L413).

**Comment by jcannon-gplates** [comment]
There's new functionality from #316 (using recent pygplates 1.0.0rc1 functionality) that affects this issue. However the trench plate IDs are still not quite correct yet (the new functionality, unlike the original functionality, uses overriding plate ID) as noted here: https://github.com/GPlates/gplately/blob/ca4a898b8ba7ae727d72b71b801f46098af22acb/gplately/reconstruction.py#L665-L673

Just a note to myself: The fix mentioned in that TODO has now been added to pygplates and will be available in pygplates 1.0.0 (soon). So I'll update gplately to use the trench ID when that is officially released (soon), and will also rename the column to "trench plate ID" to close this issue.
